jupyterhub/tests/selenium/test_browser.py

```diff
@@ -209,36 +225,686 @@ async def test_open_url_login(
         ("user", "password"),
     ],
 )
-async def test_invalid_credantials(app, browser, user, pass_w):
-    await open_url(app, browser)
+async def test_login_with_invalid_credantials(app, browser, user, pass_w):
+    await open_url(app, browser, param="login")
     await login(browser, user, pass_w)
-    await asyncio.sleep(0.1)
+    await asyncio.sleep(0.5)
     """adding for a catching of the reflected error"""
     try:
         error = browser.find_element(*LoginPageLocators.ERROR_INVALID_CREDANTIALS)
         await webdriver_wait(browser, EC.visibility_of(error))
     except NoSuchElementException:
         error = None
 
-    # verify error message and url still eguals to the login page
+    # verify error message and url
     assert LoginPageLocators.ERROR_MESSAGES_LOGIN == error.text
     assert 'hub/login' in browser.current_url
 
 
-# HOME PAGE
-async def open_home_page(app, browser, user="test_user", pass_w="test_user"):
+# MENU BAR
+
+
+@pytest.mark.parametrize(
+    "value, user, pass_w",
+    [
+        # the home page: verify if links work on the top bar
+        ("/hub/home", "test_user", "test_user"),
+        # the token page: verify if links work on the top bar
+        ("/hub/token", "test_user", "test_user"),
+        # the spawn-pending page: verify if links work on the top bar
+        ("hub/spawn", "test_user", "test_user"),
+        # "hub/not" = any url that is not existed: verify if links work on the top bar
+        ("hub/not", "test_user", "test_user"),
+        # the login page: verify if links work on the top bar
+        ("", None, None),
+    ],
+)
+async def test_menu_bar(app, browser, slow_spawn, value, user, pass_w):
```

Comment:
Probably don't need parametrize on `user, pass_w` if it's always the same. Maybe a True/False for logged-in?

Let's also use a more informative name for `value`, maybe `page`?